packages/boot/test/fast-usdc/snapshots/fast-usdc.test.ts.md

```diff
@@ -161,14 +161,14 @@ Generated by [AVA](https://avajs.dev).
 
 ## makes usdc advance
 
-> Under "published", the "fastUsdc.status" node is delegated to the statuses of fast USDC transfers identified by their tx hashes.
+> Under "published", the "fastUsdc.txns" node is delegated to the Ethereum transactions upon which Fast USDC is acting.
 > The example below illustrates the schema of the data published there.
 > 
 > See also board marshalling conventions (_to appear_).
 
     [
       [
-        'published.fastUsdc.status.0xc81bc6105b60a234c7c50ac17816ebcd5561d366df8bf3be59ff387552761702',
-        'ADVANCING',
```

Comment:
Question: I'm a bit confused. Below we have:
```
  t.deepEqual(
    storage.getPureData(`fun.txns.${mockEvidence.txHash}`),
    [mockEvidence, { status: PendingTxStatus.Advancing }],
    'tx is Advancing',
  );
  ```
  I see the `{ status: PendingTxStatus.Advancing }` part here, but not the `mockEvidence`. Wouldn't that be expected here?

Reply:
I think this one is truncating the array to show the HEAD value. Do you want it to have the whole sequence? That's only per block and in practice the state changes would happen in different blocks but the fake vstorage treats everything as block height zero

Reply:
It appears from looking at `status-manager.js` that the evidence is written in the same block as the status https://github.com/Agoric/agoric-sdk/blob/33173f9535005998b7ea7f44aa6ac3b81ba09085/packages/fast-usdc/src/exos/status-manager.js#L161-L162. I think the `sequence: true` when creating the node ensures they're both included.

I guess this test is just looking at the last cell, but if it's important behavior that the evidence is included in the first cell, I think we should capture that in the test somehow. Why not include it in all cells like `{ evidence: {}, status: "" }`? Are we assuming the indexer will just pick up the evidence in the first cell and save it?

Reply:
>Why not include it in all cells like { evidence: {}, status: "" }?

Great idea. I'll do that.

>Are we assuming the indexer will just pick up the evidence in the first cell and save it?

The indexer will accumulate events to update the db row, yeah.

Reply:
I was thinking of including it in all cells so we wouldn't have to track anything but the latest, it's kind of moot if we only include the evidence in the first cell, even though it's one less cell to track it's still more than one. However, it looks like [documentStorageSchema](https://github.com/Agoric/agoric-sdk/blob/5939b2a0fc3eb288822df07497b0b1891c0791f1/packages/internal/src/storage-test-utils.js#L258) doesn't support multiple cells. That probably doesn't have to be in scope, but a regular assert for the original cell with the evidence would suffice. I think it's important to verify that the evidence is published since the indexer relies on it.

Reply:
> I was thinking of including it in all cells so we wouldn't have to track anything but the latest,

Oh, the evidence isn't stored in a way that it can be looked up for each update.

>I think it's important to verify that the evidence is published since the indexer relies on it.

I think that's covered by the unit tests but I've added the assertion to the boot test too. Do you think the A3P tests also need it?

Reply:
> Oh, the evidence isn't stored in a way that it can be looked up for each update.

Ah I see, yea not worth changing that part.

> I think that's covered by the unit tests

I guess that makes sense. Unit tests for exhaustiveness, and a boot test just to make sure the vstorage write works for any path. It was confusing seeing the different results between the two though so thanks for updating.

> Do you think the A3P tests also need it?

I'm not sure the scope of the a3p tests, seems like they're not super important for fast-usdc because they're on a single chain, but the multichain-testing tests might want to have test cases for the status updates. There weren't any such existing test cases there before this PR though so I don't think it's in scope.